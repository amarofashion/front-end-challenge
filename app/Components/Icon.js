import React from 'react'

const Icon = ({name}) => {
  const iconName = `icon lnr lnr-${name}`
  const idName = `#lnr-${name}`

  return (
    <svg className={iconName}>
      <use xlinkHref={idName}></use>
    </svg>
  )
}

export default Icon
