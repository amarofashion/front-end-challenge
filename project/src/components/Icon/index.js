import React from 'react';
import { string } from 'prop-types';
import {
  IoIosInformationCircleOutline,
  IoIosChatbubbles,
  IoIosMail,
  IoMdClose,
} from 'react-icons/io';
import {
  FaWhatsapp,
  FaUserCircle,
  FaRegStar,
  FaStar,
  FaStarHalfAlt,
  FaRegHeart,
  FaChevronDown,
  FaChevronLeft,
} from 'react-icons/fa';
import { FiPlusCircle } from 'react-icons/fi';

const defaultProps = {
  color: '',
  size: '',
  className: '',
};

const propTypes = {
  name: string.isRequired,
  className: string,
  color: string,
  size: string,
};

export const iconsLibrary = {
  info: IoIosInformationCircleOutline,
  whatsapp: FaWhatsapp,
  user: FaUserCircle,
  plus: FiPlusCircle,
  emptyStar: FaRegStar,
  fullStar: FaStar,
  halfStar: FaStarHalfAlt,
  chat: IoIosChatbubbles,
  mail: IoIosMail,
  heart: FaRegHeart,
  arrowDown: FaChevronDown,
  arrowLeft: FaChevronLeft,
  close: IoMdClose,
};

function Icon({ name, color, size, className }) {
  const IconComponent = iconsLibrary[name];

  return (
    <IconComponent
      color={color}
      size={size}
      name={name}
      className={className}
    />
  );
}

Icon.defaultProps = defaultProps;
Icon.propTypes = propTypes;

export default Icon;
