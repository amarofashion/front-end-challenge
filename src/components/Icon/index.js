import React from 'react';
import { string } from 'prop-types';
import {
  FiShoppingBag,
  FiSearch,
  FiXCircle,
  FiChevronsDown,
  FiX,
} from 'react-icons/fi';

const defaultProps = {
  color: '',
  size: '',
  className: '',
};

const propTypes = {
  name: string.isRequired,
  className: string,
  color: string,
  size: string,
};

export const iconsLibrary = {
  cart: FiShoppingBag,
  search: FiSearch,
  close: FiXCircle,
  down: FiChevronsDown,
  delete: FiX,
};

function Icon({ name, color, size, className }) {
  const IconComponent = iconsLibrary[name];

  return (
    <IconComponent
      color={color}
      size={size}
      name={name}
      className={className}
    />
  );
}

Icon.defaultProps = defaultProps;
Icon.propTypes = propTypes;

export default Icon;
