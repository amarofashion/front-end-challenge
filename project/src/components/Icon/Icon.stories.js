import React from 'react';
import { storiesOf } from '@storybook/react';

import Icon, { iconsLibrary } from '.';

const stories = storiesOf('Icon', module);

Object.keys(iconsLibrary).map(item =>
  stories.add(`${item}`, () => <Icon name={item} />)
);
