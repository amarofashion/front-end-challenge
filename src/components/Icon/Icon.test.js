import React from 'react';
import { mount } from 'enzyme';

import Icon from '.';

describe('<Icon />', () => {
  let wrapper;

  describe('with default props', () => {
    beforeAll(() => {
      wrapper = mount(<Icon name='cart' />);
    });

    it('shoud match snapshot with default props', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
