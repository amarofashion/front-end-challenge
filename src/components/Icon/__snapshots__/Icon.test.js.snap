// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> with default props shoud match snapshot with default props 1`] = `
<Icon
  className=""
  color=""
  name="cart"
  size=""
>
  <FiShoppingBag
    className=""
    color=""
    name="cart"
    size=""
  >
    <IconBase
      attr={
        Object {
          "fill": "none",
          "stroke": "currentColor",
          "strokeLinecap": "round",
          "strokeLinejoin": "round",
          "strokeWidth": "2",
          "viewBox": "0 0 24 24",
        }
      }
      className=""
      color=""
      name="cart"
      size=""
    >
      <svg
        color=""
        fill="none"
        height="1em"
        name="cart"
        size=""
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 2L3 6v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V6l-3-4z"
          key="0"
        />
        <line
          key="1"
          x1="3"
          x2="21"
          y1="6"
          y2="6"
        />
        <path
          d="M16 10a4 4 0 0 1-8 0"
          key="2"
        />
      </svg>
    </IconBase>
  </FiShoppingBag>
</Icon>
`;
